proto/viam/app/v1/app.proto

```diff
@@ -822,6 +822,18 @@ message FragmentHistoryEntry {
   AuthenticatorInfo edited_by = 4 [(tagger.v1.tags) = "bson:\"edited_by\" json:\"edited_by\""];
 }
 
+message FragmentVersion {
+  string version = 1;
```

Comment:
@michaellee1019 we talked about this in a dev jam before the holidays: you had some thoughts on aligning our language with registry modules. What terms do we use there, and do you have any thoughts about what we should use for fragments?

I remember you mentioning "version", "tag", and "revision" as different terms

Reply:
Great point, thanks for reminding me. I think here we'd keep the terms "version" and "tag". Whenever we want to combine the two into one field, Such as A) in the robot part config or B) in the metrics storage and protos, we should use the "revision" term. We can come up with a better name too, but something like "version_or_tag" should be avoided I think.

Reply:
i thought version was the name for revision_or_tag?

Reply:
Oops. You are right.
Revision: The fragment is saved and given an auto-incremented revision number
Tag: A string name that is a pointer to a Revision.
Version: A reference to either a Revision number or Tag string.

Reply:
TLDR: I think the following protos use the right terminology. The rest of the changes can stay as is because the version field is either referencing a revision or a tag:

```
message FragmentRevision {
  string revision = 1;
  google.protobuf.Timestamp created_at = 2;
}

message FragmentTag {
  string tag = 1;
  string revision = 2;
  google.protobuf.Timestamp created_at = 3;
  google.protobuf.Timestamp updated_at = 4;
}
```